lib/ansible/modules/windows/win_data_deduplication.ps1

```diff
@@ -0,0 +1,174 @@
+#!powershell
+
+# Copyright: 2019, rnsc(@rnsc) <github@rnsc.be>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt
+
+#Requires -Module Ansible.ModuleUtils.Legacy
+
+$ErrorActionPreference = "Stop"
+
+$params = Parse-Args $args -supports_check_mode $false
+
+$drive_letter = Get-AnsibleParam -obj $params -name 'drive_letter' -type 'str' -failifempty $true
+$state = Get-AnsibleParam -obj $params -name 'state' -type 'str' -default 'enabled'
+$settings = Get-AnsibleParam -obj $params -name 'settings' -type 'list' -default $null
+$dedup_job = Get-AnsibleParam -obj $params -name 'dedup_job' -type 'list' -default $null
+
+# In theory it works on Windows 2012, but this was only developed to work
+# from 2012R2 and higher.
+if ([System.Environment]::OSVersion.Version -lt [Version]"6.3.9600.0") {
```

Comment:
you would be better off just having `[Version]"6.3"` and not worrying about the minor number. We also have an `#AnsibleRequires -OSVersion x.y` you can add at the top of the script that does this for you automatically.